### ISSUE TITLE
Improve contrast ratio for button and link colors in new design system

### ISSUE THREAD

**Issue by lunars97** [digitalfabrik/integreat-app#3614]
### Describe the Problem

Currently there are some contrast errors from WAVE (Web Accessibility Evaluation Tools) when it comes to buttons and links in a dark/contrast mode. 
UPDATE: Seems like there is also a contrast issue on buttons in system mode as well

### Describe Your Preferred Solution

Make it pass three AAA levels of conformance when it comes to contrast of buttons and links (it should be > 4.5:1 to pass all AAA levels) to make user experience more accessible and comfortable

### Describe Possible Alternatives

Leave as it is

### Additional Information

See examples with warnings in dark mode

<img width="363" height="249" alt="Image" src="https://github.com/user-attachments/assets/40ad0f1a-a98e-45c4-a6aa-29693a4c000f" />

<img width="407" height="300" alt="Image" src="https://github.com/user-attachments/assets/b9bb21e7-dce3-498c-9c77-ac9bc0305774" />

See error warning in system mode on buttons

<img width="368" height="89" alt="Image" src="https://github.com/user-attachments/assets/2d25c23e-f0ca-45e2-80b3-2e17eef6a713" />

<img width="342" height="310" alt="Image" src="https://github.com/user-attachments/assets/5e87f43e-e63c-44f4-98b5-672764322fd3" />
